src/lib.rs

```diff
@@ -130,6 +127,16 @@ impl<C: Context> IntoContext for Report<C> {
 
 pub trait ResultIntoContext: ResultExt {
     fn into_ctx<C2: Reportable>(self) -> Result<Self::Ok, Report<C2>>;
+    // Resut::and_then
+    fn then_ctx<U, F, C2>(self, op: F) -> Result<U, Report<C2>>
```

Comment:
Per @jsgoyette's suggestion
```suggestion
    fn and_then_ctx<U, F, C2>(self, op: F) -> Result<U, Report<C2>>
```